SYSTEM_TESTS.md

```diff
@@ -0,0 +1,349 @@
+
+
+
+## Actions
+
+Actions are functions that allow us to perform a certain operation on the element. Logic, right? :smile:
+
+
+
+**The ones that you should now:**
+
+- `attach_file` - attach a file to an element
+- `fill_in` - fill in the input/textarea elements
+
+```ruby
+test 'example test' do
+    # Uploads an image
+    attach_file("input#profile-image", '/path/to/image')
+    
+    # Fill in input/textarea element that has an ID, NAME or LABEL TEXT of 'username'
+    # It doesn't accept a css selector like #username, the biggest caveat of using this.
+    # An alternative is find("#username").set("Vlado") -> check out the section below
+    fill_in("username", "Vlado")
+end
+```
+
+
+
+Learn more about [Actions](https://rubydoc.info/github/jnicklas/capybara/master/Capybara/Node/Actions)
+
+
+
+## Finders
+
+Finders are used to select elements from a web page. In this section, you'll learn how to use them.
+
+
+
+### find 'selector'
+
+The `find` function is one of the most powerful system tests function. It allows us to select a **single element** on the page just like we would do it in CSS/JavaScript. Any valid CSS/JavaScript selector is a valid `find` selector.
+
+Isn't that cool?
+
+```ruby
+test 'example test' do
+   # Let's select some elements
+   title = find("#heading-primary")
+   refresh_btn = find("#btn-refresh")
+   message_input = find("#form-input-message")
+    
+   # The title variable holds Capybara::Node::Element object
+   # We can do various things with it :)
+    
+   title.text # Get the title's text
+   refresh_btn.click() # Click on the button
+   message_input.set("Great post!") # Fill the input element
+    
+   # Simulate CTRL + Enter key combination
+   message_input.native.send_keys([:control, :return])
+    
+   # Simulate a single keypress (Enter in our case)
+   message_input.native.send_keys(:return)
+    
+   # More advanced element selection
+    
+   # Select the first element with a class of 'comment'
+   find(".comment", match: :first)
+    
+   # OR you can do this:
+   first(".comment")
+   
+   # Select the element that **contains** certain text
+   find("#btn-refresh", text: "Refresh")
+    
+   # Select the element that has the **exact same** text
+   find("#btn-submit", exact_text: "Submit")
+end
+```
+
+> IMPORTANT NOTE: If `find` selects multiple elements, the test fails. If you are expecting multiple elements you should either use `all` or provide 'match' argument to select a specific element.
+
+
+
+I hope you see now why `find` is one of my favourite functions when I need to select an element. Its possibilities are endless.  If this doesn't cover your case, you can always check out the [official docs](https://rubydoc.info/github/jnicklas/capybara/master/Capybara%2FNode%2FFinders:find).
```

Comment:
Maybe we can make this part less personal with something like this: "`find` is an incredibly useful function to select elements and it is a crucial component for system testing."

Reply:
:+1: